teps/0057-windows-support.md

```diff
@@ -0,0 +1,154 @@
+---
+status: proposed
+title: Windows support
+creation-date: '2021-03-18'
+last-updated: '2021-03-18'
+authors:
+- '@DrWadsy'
+- '@lukehb'
+- '@aiden-deloryn'
+---
+
+# TEP-0057: Windows support
+
+<!--
+A table of contents is helpful for quickly jumping to sections of a TEP and for
+highlighting any additional information provided beyond the standard TEP
+template.
+
+Ensure the TOC is wrapped with
+  <code>&lt;!-- toc --&rt;&lt;!-- /toc --&rt;</code>
+tags, and then generate with `hack/update-toc.sh`.
+-->
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Requirements](#requirements)
+- [Proposal](#proposal)
+  - [Notes/Caveats (optional)](#notescaveats-optional)
+  - [Risks and Mitigations](#risks-and-mitigations)
+  - [User Experience (optional)](#user-experience-optional)
+  - [Performance (optional)](#performance-optional)
+- [Design Details](#design-details)
+- [Test Plan](#test-plan)
+- [Design Evaluation](#design-evaluation)
+  - [Reusability](#reusability)
+  - [Simplicity](#simplicity)
+  - [Flexibility](#flexibility)
+  - [Conformance](#conformance)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Infrastructure Needed (optional)](#infrastructure-needed-optional)
+- [Upgrade &amp; Migration Strategy (optional)](#upgrade--migration-strategy-optional)
+- [References (optional)](#references-optional)
+<!-- /toc -->
+
+## Summary
+The current state of Windows containers requires that they be run on Windows nodes, it is not possible to run Windows containers on a Linux system. For use cases that require the use of Windows containers there is no choice but to run those workloads on Windows nodes. The current state of Tekton does not support Windows nodes in any form, and so there is currently no way to run Tekton Pipelines to support Windows workloads. The goal of this TEP is to provide support for Windows workloads within Tekton Pipelines, meaning that support for Windows nodes will be added. In addition to supporting Windows nodes within Tekton Pipelines we would also need to ensure that all Tekton components (the Operator, Triggers, the CLI/Dashboard and the Catalog/Hub) support, or reflect, Tasks which require Windows nodes.
+
```

Comment:
i think it makes sense to include these are requirements; i would say that (ideally) these components should support the Tasks which require Windows nodes _without_ requiring any changes (i.e. these components should all be decoupled enough that they don't need updates in order to support this)

possible exceptions:
* maybe the operator, if it needs to be aware of when Tasks that support windows should be installed (not sure it does any Task installation tho)
* if we were to update more components to run on windows (i.e. controllers, cli, dashboard) but that's outside the scope of this TEP anyway

Reply:
I think that's definitely what we'd be aiming for - current elements should keep working as is, with no changes necessary, other thank the elements which need to handle the Windows nodes. Everything else should just keep ticking as it already is.